Document agentless Feature Flags delivery for Node.js (#38231)
DataDog/documentation: content/en/feature_flags/concepts/configuration_sources.md

| SDK | Minimum version |
|---|---|
| Node.js `dd-trace` v5 | 5.116.0 |
| Node.js `dd-trace` v6 | 6.5.0 |

Configure only the API key and environment in the application process:

{{< code-block lang="bash" >}}
DD_API_KEY=<DATADOG_API_KEY>
DD_ENV=<YOUR_ENVIRONMENT>
{{< /code-block >}}

Then initialize or access the Datadog OpenFeature provider in application code. See the [Node.js setup instructions][2].

No configuration-source or provider-enable setting is required. Polling begins only when application code initializes or accesses the provider; installing or initializing the tracer alone does not create Feature Flags CDN traffic.

<div class="alert alert-warning">The initial Node.js agentless releases support configuration delivery and local flag evaluation only. They do not support evaluation metrics, exposure logging, or experimentation use cases.</div>

## Advanced configuration

### Choose a configuration source

| Mode | Configuration delivery | Activation | Agent requirement |
|---|---|---|---|

> | `agentless` (default) | The SDK periodically fetches UFC from the Datadog-managed CDN over HTTPS. | Polling begins when application code initializes or accesses the Datadog OpenFeature provider. | No Datadog Agent is required for flag configuration. |

Author: I'd like to discuss latency and availability here, or somewhere in this document. We should not mention the CDN vendor as it might change but suffice to say they operate over 60 global locations, include dense clusters in north america, europe and india (south east asia?)

Many of the CDN locations are peered with popular cloud vendors like GCP, Azure and AWS so we expect network traffic in those locations to not egress (double check me here): https://www.fastly.com/network-map

High availability is offered; outage of individual locations falls back to nearby locations.

Latency wise we expect typical sizes flagging deployments to receive bytes in <50ms.

Author: Addressed in [b18c59f489aadfdff1a4e279836605130e5dc967](https://github.com/DataDog/documentation/commit/b18c59f489aadfdff1a4e279836605130e5dc967): added vendor-neutral guidance covering global CDN distribution, network peering, likely geographic proximity, and redundant routing without naming the vendor or asserting latency or egress guarantees.

| `remote_config` | The Datadog Agent receives UFC through Remote Configuration and delivers it to the SDK. | Selecting `remote_config` enables the Feature Flags Remote Configuration subscription. | Requires an Agent with Remote Configuration enabled. |

Set `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE` only when you want to select a source explicitly:

{{< code-block lang="bash" >}}
DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=agentless
{{< /code-block >}}

The SDK resolves the source once during initialization. Restart the application to change sources.

> ### Configure Agentless delivery

Author: Addressed in [4df3f4876b2359d8a2de96b0344c555787ec8da6](https://github.com/DataDog/documentation/commit/4df3f4876b2359d8a2de96b0344c555787ec8da6): Changed the heading to sentence case.

- polls every 30 seconds by default;
- uses a 5-second request timeout by default;
- uses ETags to avoid downloading unchanged configuration;
- preserves the last accepted configuration during temporary network or payload errors; and
- prevents overlapping polls.

Keep `DD_API_KEY` in a secret manager and expose it only to the application process that loads flag configuration. Agentless configuration delivery sends the API key directly from the application to Datadog over HTTPS.

#### Use a custom agentless endpoint

The Datadog-managed endpoint is recommended for standard deployments. For power-user testing, local development, or an operator-managed proxy, override it with `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_BASE_URL`:

{{< code-block lang="bash" >}}
DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_BASE_URL=http://localhost:8080
{{< /code-block >}}

The custom URL must use HTTP or HTTPS. If it contains only an origin or a root path, the SDK appends the canonical UFC path. If it contains a non-root path, the SDK uses that path as the complete endpoint.

> `DD_API_KEY` is required for the default Datadog-managed endpoint. A custom endpoint can start without it. If `DD_API_KEY` is configured, the SDK forwards it to the custom endpoint, including over HTTP. Prefer HTTPS, and do not configure the Datadog API key for a cleartext endpoint unless you control and trust the network path.

Author: > If `DD_API_KEY` is configured, the SDK forwards it to the custom endpoint, including over HTTP.

No, we decided to strip it out in https://github.com/DataDog/dd-trace-js/pull/9481 (see feedback) and decisions:

Ruben and I discussed the path forward:

in misconfiguration situation, not throwing but still defaulting to provider off, logging loudly is canonical 
Local development support: continue to ease http restrictions but fully strip out dd api key, only sending it over https to our first party api

Author: Addressed in [4df3f4876b2359d8a2de96b0344c555787ec8da6](https://github.com/DataDog/documentation/commit/4df3f4876b2359d8a2de96b0344c555787ec8da6): Updated the custom endpoint contract: the API key is sent only over HTTPS to the Datadog-managed endpoint, never to a custom endpoint; invalid custom endpoint settings disable the provider, log the error, and return caller-provided defaults.

| Explicit `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=agentless` | Selects CDN delivery. Polling begins when application code initializes or accesses the provider. |
| Explicit `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=remote_config` | Selects Agent delivery and enables the Feature Flags Remote Configuration subscription. |
| Blank or whitespace-only `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE` | Treats the source as unset, so the legacy migration setting or the agentless default applies. |
| Explicit `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=offline` or another unsupported nonblank value | Fails closed when application code accesses the provider. The SDK does not select CDN or Remote Configuration delivery. |
| No source and `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED=true` | Preserves Remote Configuration during the migration window. |
| No source and `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED=false` | Keeps the provider and both delivery paths disabled. |
| Neither setting is present | Selects agentless delivery. Polling begins when application code initializes or accesses the provider. |

> ### Use Agent remote configuration

Author: Addressed in [4df3f4876b2359d8a2de96b0344c555787ec8da6](https://github.com/DataDog/documentation/commit/4df3f4876b2359d8a2de96b0344c555787ec8da6): Capitalized the Remote Configuration product name.

Set the source to `remote_config` to use Agent-managed delivery:

{{< code-block lang="bash" >}}
DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=remote_config
{{< /code-block >}}

Configure the API key on the Agent, not in the application process. If Remote Configuration has been disabled on the Agent, re-enable it. See [Remote Configuration][1] for Agent setup and network requirements.

> ### Migrate an existing remote configuration setup

Author: Addressed in [4df3f4876b2359d8a2de96b0344c555787ec8da6](https://github.com/DataDog/documentation/commit/4df3f4876b2359d8a2de96b0344c555787ec8da6): Capitalized the Remote Configuration product name.

Existing customers who set `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED=true` remain on Remote Configuration during a migration window. This deprecated setting is a compatibility bridge, not the long-term configuration.

When you are ready to use agentless delivery:

1. Set `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=agentless`.
2. For the Datadog-managed endpoint, configure `DD_API_KEY` and `DD_ENV` in the application, and set `DD_SITE` if your organization is not on the default site.
3. Initialize the provider and verify that it receives flag updates.
4. Remove `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED`. Do not leave the deprecated setting enabled after the migration is complete.

To remain on Agent Remote Configuration temporarily, set `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=remote_config`, then remove `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED`. Your API key remains on the Agent.

If you set `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED=false`, replace it with `DD_FEATURE_FLAGS_ENABLED=false`.

> Explicit `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE` values take precedence over the legacy setting. After the legacy setting is removed, a Node.js application without an explicit source uses agentless delivery. Set `remote_config` explicitly before that release if you want to remain on Agent delivery.

Author: Addressed in [4df3f4876b2359d8a2de96b0344c555787ec8da6](https://github.com/DataDog/documentation/commit/4df3f4876b2359d8a2de96b0344c555787ec8da6): Reworded the migration trigger to the removal of the deprecated legacy setting.

### Future offline mode

> Offline mode is planned for applications that provide UFC JSON bytes at startup and do not make CDN or Remote Configuration network requests. It is reserved but not implemented in the initial agentless releases. Setting the source to `offline` fails closed without selecting another delivery path.

Author: UFC is not known, flag configuration

Author: Addressed in [4df3f4876b2359d8a2de96b0344c555787ec8da6](https://github.com/DataDog/documentation/commit/4df3f4876b2359d8a2de96b0344c555787ec8da6): Replaced UFC JSON with flag configuration JSON.

| Environment variable | Default | Description |
|---|---|---|
| `DD_SITE` | `datadoghq.com` | Datadog site used to derive the agentless endpoint. |
| `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_BASE_URL` | Datadog-managed endpoint | Overrides the agentless flag configuration endpoint or base URL. See [Use a custom agentless endpoint](#use-a-custom-agentless-endpoint). |
| `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_POLL_INTERVAL_SECONDS` | `30` | Positive integer that sets the time between completed polling attempts, capped at 3600 seconds. |
| `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_REQUEST_TIMEOUT_SECONDS` | `5` | Positive integer that sets the timeout for an individual configuration request. |

The SDK fetches configuration in the background and evaluates flags locally. Individual flag evaluations do not make network requests. The agentless source:

- polls every 30 seconds by default;
- uses a 5-second request timeout by default;
- uses ETags to avoid downloading unchanged configuration;
- preserves the last accepted configuration during temporary network or payload errors; and
- prevents overlapping polls.

> The Datadog-managed CDN has points of presence across major global regions and peers with internet service providers and content networks. This makes a serving location likely to be geographically close to most application workloads. Multiple locations and redundant interconnections also allow the CDN to route around unavailable network paths or locations.

Author: Addressed in [dd6638fbc8b52d5c66c75fc9873cc3f33aac8643](https://github.com/DataDog/documentation/commit/dd6638fbc8b52d5c66c75fc9873cc3f33aac8643): tightened the paragraph while retaining vendor-neutral global distribution, peering, redundant routing, and likely proximity guidance.

Keep `DD_API_KEY` in a secret manager and expose it only to the application process that loads flag configuration. Agentless configuration delivery sends the API key directly from the application to Datadog over HTTPS.

#### Use a custom agentless endpoint

The Datadog-managed endpoint is recommended for standard deployments. For power-user testing, local development, or an operator-managed proxy, override it with `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_BASE_URL`:

{{< code-block lang="bash" >}}
DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_BASE_URL=http://localhost:8080
{{< /code-block >}}

> The custom URL must use HTTP or HTTPS. If it contains only an origin or a root path, the SDK appends the standard flag configuration path. If it contains a non-root path, the SDK uses that path as the complete endpoint.

Reviewer: More specifically, HTTP is only allowed for local hosts for testing purposes.

Reviewer: 🤖 **Response from AI:** Addressed in a45a054a65 — the custom-endpoint section now states that HTTP is permitted only when the URL targets a loopback host; any other custom endpoint must use HTTPS.

> The SDK sends `DD_API_KEY` only over HTTPS to the default Datadog-managed endpoint. It never forwards the Datadog API key to a custom endpoint. Custom endpoints can use HTTP for controlled local development; use HTTPS for any endpoint outside the local development environment.

Reviewer: I think we do forward to custom endpoints, just never over HTTP connections, only HTTPS

Reviewer: 🤖 **Response from AI:** Addressed in a45a054a65 — updated to say the SDK sends the `DD-API-KEY` header to custom endpoints too (reversing the earlier "never"), matching shipped `dd-trace` 5.116.0/6.5.0. Heads-up: dd-trace-js #9481 (still open) would strip the key for custom endpoints — if it ships in this release train, this section should flip back.

If the custom endpoint setting is invalid, the SDK keeps the provider disabled, logs the configuration error, and evaluations return caller-provided default values.

Datadog-managed agentless delivery is not available for Datadog for Government in these Node.js versions. Applications on that site continue to use caller-provided default values unless they use Agent Remote Configuration.

### Enable or disable feature flags

> `DD_FEATURE_FLAGS_ENABLED` defaults to `true`, so new setups do not need to set it. Set it to `false` to disable the provider and both configuration delivery paths:

Reviewer: Perhaps we should highlight that disabling the provider will result in the code-provided defaults being served for all assignment requests, with a `STATIC` reason code.

{{< code-block lang="bash" >}}
DD_FEATURE_FLAGS_ENABLED=false
{{< /code-block >}}

### Activation and billing

Server Feature Flags billing is based on configuration requests made through Remote Configuration or the CDN. Installing the tracer does not activate either delivery path by itself.

- With the default agentless source, CDN polling starts only when application code initializes or accesses the Datadog OpenFeature provider.

> - Explicitly selecting `remote_config` starts the Agent Feature Flags subscription. It does not require application code to initialize the provider.

Reviewer: ❤️  Appreciate the clarity

> ### Configuration precedence

Reviewer: 💪

| Configuration | Result |
|---|---|
| `DD_FEATURE_FLAGS_ENABLED=false` | Disables the provider and both delivery paths, regardless of other settings. |
| Explicit `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=agentless` | Selects CDN delivery. Polling begins when application code initializes or accesses the provider. |
| Explicit `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=remote_config` | Selects Agent delivery and enables the Feature Flags Remote Configuration subscription. |
| Blank or whitespace-only `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE` | Treats the source as unset, so the legacy migration setting or the agentless default applies. |
| Explicit `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=offline` or another unsupported nonblank value | Fails closed when application code accesses the provider. The SDK does not select CDN or Remote Configuration delivery. |
| No source and `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED=true` | Preserves Remote Configuration during the migration window. |
| No source and `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED=false` | Keeps the provider and both delivery paths disabled. |

> | Neither setting is present | Selects agentless delivery. Polling begins when application code initializes or accesses the provider. |

Author: Addressed in [dd6638fbc8b52d5c66c75fc9873cc3f33aac8643](https://github.com/DataDog/documentation/commit/dd6638fbc8b52d5c66c75fc9873cc3f33aac8643): named both configuration settings explicitly in the precedence row.

### Use Agent Remote Configuration

Set the source to `remote_config` to use Agent-managed delivery:

{{< code-block lang="bash" >}}
DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=remote_config
{{< /code-block >}}

Configure the API key on the Agent, not in the application process. If Remote Configuration has been disabled on the Agent, re-enable it. See [Remote Configuration][1] for Agent setup and network requirements.

> ### Migrate an existing Remote Configuration setup

Reviewer: 🙌

---
title: Server SDK Configuration Sources
description: Understand how Datadog Feature Flags server SDKs receive flag configuration.
further_reading:
- link: "/feature_flags/server/"
  tag: "Documentation"
  text: "Set up Server-Side Feature Flags"
- link: "/feature_flags/implementation_patterns/serverless/"
  tag: "Documentation"
  text: "Use Feature Flags in serverless environments"
- link: "/remote_configuration/"
  tag: "Documentation"
  text: "Learn about Remote Configuration"
---

Datadog Feature Flags [server-side SDKs][3] evaluate flags locally from flag configuration. The _configuration source_ determines how the SDK receives that configuration; it does not change OpenFeature evaluation semantics.

## Get started with agentless delivery

> Agentless delivery is the default in supported Node.js SDK versions. Other server SDKs use Agent Remote Configuration for flag delivery. To get started with Node.js, use one of these minimum versions:

Reviewer 2: ```suggestion
Agentless delivery is the default in the following supported SDKs and versions. Other server SDKs use Agent Remote Configuration for flag delivery. To get started, use one of these minimum versions:
```

| Mode | Configuration delivery | Activation | Agent requirement |
|---|---|---|---|
| `agentless` (default) | The SDK periodically fetches flag configuration from the Datadog-managed CDN over HTTPS. | Polling begins when application code initializes or accesses the Datadog OpenFeature provider. | No Datadog Agent is required for flag configuration. |
| `remote_config` | The Datadog Agent receives flag configuration through Remote Configuration and delivers it to the SDK. | Selecting `remote_config` enables the Feature Flags Remote Configuration subscription. | Requires an Agent with Remote Configuration enabled. |

Set `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE` only when you want to select a source explicitly:

{{< code-block lang="bash" >}}
DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=agentless
{{< /code-block >}}

The SDK resolves the source once during initialization. Restart the application to change sources.

### Configure agentless delivery

If your organization is not on the default `datadoghq.com` site, set `DD_SITE` in the application process. The agentless source also supports these operational settings:

| Environment variable | Default | Description |
|---|---|---|

> | `DD_SITE` | `datadoghq.com` | Datadog site used to derive the agentless endpoint. |

Reviewer 2: Does "default site" here _only_ refer to the US1 region (`datadoghq.com`), or do you mean "the default site for your region" (so for Asia customers, default might be `ap1.datadog.com`)?

If we need to account for other regions, this shortcode changes the URL based on what the reader has selected on the docs site:

<img width="500" alt="image" src="https://github.com/user-attachments/assets/c40d702f-ac9d-4bc6-b7d6-c746c9db8910" />

```suggestion
If your organization is not on the default {{< region-param key="dd_site" code="true" >}} site, set `DD_SITE` in the application process. The agentless source also supports these operational settings:

| Environment variable | Default | Description |
|---|---|---|
| `DD_SITE` | {{< region-param key="dd_site" code="true" >}} | Datadog site used to derive the agentless endpoint. |
```

Author: US1 region (datadoghq.com)

| `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_BASE_URL` | Datadog-managed endpoint | Overrides the agentless flag configuration endpoint or base URL. See [Use a custom agentless endpoint](#use-a-custom-agentless-endpoint). |
| `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_POLL_INTERVAL_SECONDS` | `30` | Positive integer that sets the time between completed polling attempts, capped at 3600 seconds. |
| `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_REQUEST_TIMEOUT_SECONDS` | `5` | Positive integer that sets the timeout for an individual configuration request. |

The SDK fetches configuration in the background and evaluates flags locally. Individual flag evaluations do not make network requests. The agentless source:

- polls every 30 seconds by default;
- uses a 5-second request timeout by default;
- uses ETags to avoid downloading unchanged configuration;
- preserves the last accepted configuration during temporary network or payload errors; and

> - prevents overlapping polls.

Reviewer 2: ```suggestion
The SDK fetches configuration in the background and evaluates flags locally. Individual flag evaluations do not make network requests. The agentless source does the following:

- Polls every 30 seconds by default
- Uses a 5-second request timeout by default
- Uses ETags to avoid downloading unchanged configuration
- Preserves the last accepted configuration during temporary network or payload errors
- Prevents overlapping polls
```

The Datadog-managed CDN uses globally distributed points of presence, network peering, and redundant routing. Serving locations are therefore likely to be geographically close to most application workloads.

Keep `DD_API_KEY` in a secret manager and expose it only to the application process that loads flag configuration. Agentless configuration delivery sends the API key directly from the application to Datadog over HTTPS.

#### Use a custom agentless endpoint

> The Datadog-managed endpoint is recommended for standard deployments. For power-user testing, local development, or an operator-managed proxy, override it with `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_BASE_URL`:

Reviewer 2: ```suggestion
The Datadog-managed endpoint is recommended for standard deployments. For advanced testing, local development, or an operator-managed proxy, override it with `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_BASE_URL`:
```

{{< code-block lang="bash" >}}
DD_FEATURE_FLAGS_CONFIGURATION_SOURCE_AGENTLESS_BASE_URL=http://localhost:8080
{{< /code-block >}}

The custom URL must use HTTP or HTTPS. If it contains only an origin or a root path, the SDK appends the standard flag configuration path. If it contains a non-root path, the SDK uses that path as the complete endpoint.

The SDK sends `DD_API_KEY` only over HTTPS to the default Datadog-managed endpoint. It never forwards the Datadog API key to a custom endpoint. Custom endpoints can use HTTP for controlled local development; use HTTPS for any endpoint outside the local development environment.

If the custom endpoint setting is invalid, the SDK keeps the provider disabled, logs the configuration error, and evaluations return caller-provided default values.

> Datadog-managed agentless delivery is not available for Datadog for Government in these Node.js versions. Applications on that site continue to use caller-provided default values unless they use Agent Remote Configuration.

Reviewer 2: ```suggestion
Datadog-managed agentless delivery is not available for Datadog for Government in the supported SDK versions. Applications on that site continue to use caller-provided default values unless they use Agent Remote Configuration.
```

When you are ready to use agentless delivery:

1. Set `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=agentless`.
2. For the Datadog-managed endpoint, configure `DD_API_KEY` and `DD_ENV` in the application, and set `DD_SITE` if your organization is not on the default site.
3. Initialize the provider and verify that it receives flag updates.
4. Remove `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED`. Do not leave the deprecated setting enabled after the migration is complete.

To remain on Agent Remote Configuration temporarily, set `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE=remote_config`, then remove `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED`. Your API key remains on the Agent.

If you set `DD_EXPERIMENTAL_FLAGGING_PROVIDER_ENABLED=false`, replace it with `DD_FEATURE_FLAGS_ENABLED=false`.

Explicit `DD_FEATURE_FLAGS_CONFIGURATION_SOURCE` values take precedence over the legacy setting. After the legacy setting is removed, a Node.js application without an explicit source uses agentless delivery. Set `remote_config` explicitly before the deprecated legacy setting is removed if you want to remain on Agent delivery.

### Future offline mode

> Offline mode is planned for applications that provide flag configuration JSON at startup and do not make CDN or Remote Configuration network requests. It is reserved but not implemented in the initial agentless releases. Setting the source to `offline` fails closed without selecting another delivery path.

Reviewer 2: Remove this section entirely; docs should only cover the product's currently functionality, not try to predict the future.
```suggestion
```

further_reading:
- link: "/feature_flags/server/"
  tag: "Documentation"
  text: "Set up Server-Side Feature Flags"
- link: "/feature_flags/implementation_patterns/serverless/"
  tag: "Documentation"
  text: "Use Feature Flags in serverless environments"
- link: "/remote_configuration/"
  tag: "Documentation"
  text: "Learn about Remote Configuration"
---

Datadog Feature Flags [server-side SDKs][3] evaluate flags locally from flag configuration. The _configuration source_ determines how the SDK receives that configuration; it does not change OpenFeature evaluation semantics.

## Choose a configuration source

These configuration source options are available for supported server SDKs:

`agentless`
: The SDK periodically fetches flag configuration from the Datadog-managed CDN over HTTPS.
  - Polling begins when application code initializes or accesses the Datadog OpenFeature provider.
  - No Datadog Agent is required for flag configuration.

`remote_config`
: The Datadog Agent receives flag configuration through Remote Configuration and delivers it to the SDK.
  - Selecting `remote_config` enables the Feature Flags Remote Configuration subscription.

>   - Requires a Datadog Agent with Remote Configuration enabled.

Reviewer 2: This content was dense and awkward to read as a 4-column table, so I tried [reformatting](https://github.com/DataDog/documentation/pull/38231/commits/b6a862b57eb18830a390f36cc35d125d9d2eb5f0) as a description list. Happy to keep iterating if needed.

<img width="600" alt="image" src="https://github.com/user-attachments/assets/d02b354c-8494-4219-8b24-d03b0be499c6" />
